app/controller/button/SpatialQueryButton.js

```diff
@@ -0,0 +1,157 @@
+
+    /**
+     * Function to determine the query layer if not yet defined in class
+     */
+    findQueryLayer: function () {
+        var me = this;
+        var view = me.getView();
+        if (!view.queryLayer && view.queryLayerName) {
+            view.queryLayer = BasiGX.util.Layer.
+                getLayerByName(view.queryLayerName);
+        }
+    },
+
+    /**
+     * Activates #drawQueryInteraction on button toggle to draw polygon
+     * selection geometry that will be used for filtering.
+     *
+     * @param {Ext.button.Button} btn The toggled select by polygon button.
+     * @param {Boolean} pressed The toggle state.
+     */
+    onSpatialQueryBtnToggle: function (btn, pressed) {
+        var me = this;
+        var view = me.getView();
+
+        if (!view.queryLayer) {
+            me.findQueryLayer();
+        }
+
+        var geometryFunction;
+        var type = view.drawGeometryType;
+        if (view.spatialOperator === 'bbox') {
+            type = 'Circle';
+            geometryFunction = ol.interaction.Draw.createBox();
+        }
+
+        if (!me.drawQueryInteraction) {
+            me.drawQueryInteraction = new ol.interaction.Draw({
+                features: view.queryFeature,
+                geometryFunction: geometryFunction,
+                type: type
+            });
+            view.map.addInteraction(me.drawQueryInteraction);
+        }
+        if (pressed) {
+            me.drawQueryInteraction.setActive(true);
+            view.queryFeature.on('add', me.getGeometryFromPolygonAndTriggerWfs, me);
+        } else {
+            me.drawQueryInteraction.setActive(false);
+            view.queryFeature.un('add', me.getGeometryFromPolygonAndTriggerWfs, me);
+        }
+    },
+
+    /**
+     * Help method to create a polygon geometry from drawn irregular polygon.
+     *
+     * @param {Ext.Event} evt The add-Event containing drawn feature
+     */
+    getGeometryFromPolygonAndTriggerWfs: function (evt) {
+        var geometry = evt.element.getGeometry();
+        this.buildAndRequestQuery(geometry);
+    },
+
+    /**
+     * Build query / filter and call WFS
+     *
+     * @param {ol.geom.Geometry} geometry The geometry
+     */
+    buildAndRequestQuery: function (geometry) {
+        var me = this;
+        var view = me.getView();
+        if (!view.queryLayer) {
+            return;
+        }
+        var mapComp = BasiGX.util.Map.getMapComponent();
+        var projString = mapComp.getMap().getView().getProjection().getCode();
+        var geomFieldName = view.queryLayer.get('geomFieldName') ||
+            view.queryLayer.getSource().get('geomFieldName') ||
+            'the_geom';
+        var url = view.queryLayer.get('url') ||
+            view.queryLayer.getSource().getUrl() ||
+            view.queryLayer.getSource().getUrls()[0];
+
+        var featureType = view.queryLayer.get('featureType') ||
+            view.queryLayer.getSource().getParams()['LAYERS'];
```

Comment:
This could also be `layers` or even `laYeRs`. As far as I remember there is a util to detect the `LAYERS` param case independent. Please use this here.

Reply:
Thx for the hint!

Now I use
```
BasiGX.util.Object.layersFromParams(
    view.queryLayer.getSource().getParams()
);
```